#
# This patch has been re-diffed against kernel-source-2.4.27-8, 
# as the original one applied to it with 6 line offset. Original
# patch description follows.
#
# -- Jurij Smakov <jurij@wooyd.org>  Wed,  2 Mar 2005 20:46:14 -0500
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/02 14:12:47-08:00 davem@nuts.davemloft.net 
#   [SPARC64]: Accept 'm5823' clock chip as seen on SB1500.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# arch/sparc64/kernel/time.c
#   2005/03/02 14:12:26-08:00 davem@nuts.davemloft.net +5 -2
#   [SPARC64]: Accept 'm5823' clock chip as seen on SB1500.
# 
--- a/arch/sparc64/kernel/time.c	2003-11-28 13:26:19.000000000 -0500
+++ b/arch/sparc64/kernel/time.c	2005-03-02 20:32:12.000000000 -0500
@@ -776,6 +776,7 @@
 		    strcmp(model, "mk48t59") &&
 		    strcmp(model, "m5819") &&
 		    strcmp(model, "m5819p") &&
+		    strcmp(model, "m5823") &&
 		    strcmp(model, "ds1287")) {
 			if (cbus != NULL) {
 				prom_printf("clock_probe: Central bus lacks timer chip.\n");
@@ -835,7 +836,8 @@
 
 			if (!strcmp(model, "ds1287") ||
 			    !strcmp(model, "m5819") ||
-			    !strcmp(model, "m5819p")) {
+			    !strcmp(model, "m5819p") ||
+			    !strcmp(model, "m5823")) {
 				ds1287_regs = edev->resource[0].start;
 			} else {
 				mstk48t59_regs = edev->resource[0].start;
@@ -856,7 +858,8 @@
 			}
 			if (!strcmp(model, "ds1287") ||
 			    !strcmp(model, "m5819") ||
-			    !strcmp(model, "m5819p")) {
+			    !strcmp(model, "m5819p") ||
+			    !strcmp(model, "m5823")) {
 				ds1287_regs = isadev->resource.start;
 			} else {
 				mstk48t59_regs = isadev->resource.start;
